validphys2/src/validphys/loader.py

```diff
@@ -62,46 +62,69 @@ class CutsNotFound(LoadFailedError): pass
 
 class PDFNotFound(LoadFailedError): pass
 
+class ProfileNotFound(LoadFailedError): pass
+
 class RemoteLoaderError(LoaderError): pass
 
 class InconsistentMetaDataError(LoaderError): pass
 
+def _get_nnpdf_profile(profile_path=None):
+    """Returns the NNPDF profile as a dictionary
+    If no ``profile_path`` is provided it will be autodiscovered in the following order:
+
+    Environment variable $NNPDF_PROFILE_PATH
+    {sys.prefix}/share/NNPDF/nnprofile.yaml
+    {sys.base_prefix}/share/NNPDF/nnprofile.yaml
+
+    If no profile is found a LoaderError will be thrown
+    """
+    profile_path = os.environ.get("NNPDF_PROFILE_PATH", profile_path)
+    if profile_path is None:
+        # Check both sys paths
+        prefix_paths = [sys.prefix, sys.base_prefix]
+        for prefix in prefix_paths:
+            check = pathlib.Path(prefix) / "share/NNPDF/nnprofile.yaml"
+            if check.exists():
+                profile_path = check
+                break
```

Comment:
The corresponding one liner is
```python
next((p for prefix in  (sys.prefix, sys.base_prefix) if (p:=(pathlib.Path(prefix)/"share/NNPDF/nnprofile.yaml")).exists() ),None)
```
but seeing as I spent 10 minutes searching for the appropriate syntax for the walrus (which I find backwards) I am not really recommending it.

Reply:
I don't really one liners like that. Looks cool but it's not like one gains anything.
I like the walrus though, since we are 3.8 now I might start using it

Reply:
FWIW every time I have thought about using it I either got wrong the operator precedence or had do search the precise scoping rules, so not much of a gain overall...

Reply:
I've avoided until now because python 3.6 is still everywhere by means of old computers amd clusters so I don't have much experience with it.
But I like the fact that it's a walrus :P